docs/proposals/timeseries-partitioning-in-compactor.md

```diff
@@ -124,34 +124,27 @@ T1 - Partition 1-2 was created with hash % 2 == 0, and in order to avoid having
-5. Thanos Compactor iterates over each compaction plan (partitioned group). For each iteration, calls Planner to make sure the group is ready for compaction.
+      3. Generates compaction plan that indicates which partition ID each blocks got assigned.
+4. Grouper returns compaction plans to Compactor.
+5. Compactor iterates over each compaction plan (partitioned group). For each iteration, calls Planner to make sure the group is ready for compaction.
 6. Planner collects all compaction plans which are ready for compaction.
    1. For each compaction plan and for each blocks in the plan:
-      1. Retrieve block partition visit marker file from block-storage based on partition ID of the plan.
-      2. If there is no visit marker file or visit marker file is expired and not in completed status, put visit marker file for this block partition in block-storage. The visit marker file would be named like `partition-<partition_id>-visit-mark.json` and its content would look like:
-         ```
-         {"compactorID":"<current_compactor_id>","visitTime":"<date_now>","status":"pending","version":1}
-         ```
-      4. Otherwise, skip this compaction plan. Because at least one block partition is picked up by other compactor.
-7. Return all ready compaction plans to Thanos Compactor.
-8. Thanos Compactor starts compacting partitioned blocks from compaction plans. Once compaction completed, Thanos Compactor would upload deletion marker for the source blocks in the plan. Also, it would update `partition-<partition_id>-visit-mark.json` to have status set to `completed`.
+      1. Make sure the source block fits within the time range of the group.
+      2. Make sure the source block with assigned partition ID is currently not used by another ongoing compaction.
+   2. If all blocks in the group are ready to be compacted, mark status of those blocks along with assigned partition ID as `pending`.
+7. Return all ready compaction plans to Compactor.
+8. Compactor starts compacting partitioned blocks from compaction plans. Once compaction completed, Compactor would upload deletion marker for the source blocks in the plan. Also, it would mark status of all blocks along with assigned partition ID in the group as `completed`.
 
 ### Clean up Workflow
 
-Cleaner would periodically check any source blocks having a deletion marker. If there is a deletion marker for the block, Cleaner should retrieve compaction plan to get all partition IDs assigned to this block. If visit marker files for all partitions have status set to `completed`, this source block can be deleted. Otherwise, skip deletion.
+Cleaner would periodically check any source blocks having a deletion marker. If there is a deletion marker for the block, Cleaner should retrieve compaction plan to get all partition IDs assigned to this block. If this source block's all assigned partition ID have status set to `completed`, this source block can be deleted. Otherwise, skip deletion.
```

Comment:
Where does the cleaner get the compaction plan? Do we store it somewhere?

Reply:
Cleaner gets status from file upserted by compaction workflow above.